rumqttd/src/router/scheduler.rs

```diff
@@ -160,7 +160,9 @@ impl Tracker {
                 self.status = Status::Ready;
                 Some(previous)
             }
-            ScheduleReason::IncomingAck if previous == PauseReason::InflightFull => {
+            // There might be a case where we got IncomingAck but our inflight wasn't full
+            // So we set status to Ready in case of IncomingAck unless reason in Busy
+            ScheduleReason::IncomingAck if previous != PauseReason::Busy => {
                 self.status = Status::Ready;
                 Some(previous)
             }
```

Comment:
i am not sure big off an affect this would be but, this would cause status to be `Ready` even for incoming `PubAck`s which is not really necessary (which could lead to `connection_id` being in readyqueue multiple times). 

we could split `IncomingAck` reason to 2 parts `IncomingAck` and `IncomingAckNeedsReply` (ok no really this name xP) and only reschedule tracker when its really required. 

@mnpw wdyt?

Reply:
how does `Ready` status after getting `PubAck` cause `connection_id` to be in readyqueue multiple times?

Reply:
so where we [call `try_ready`](https://github.com/bytebeamio/rumqtt/blob/fe8a8e3c12274846965f3bbcf504706cac654d6e/rumqttd/src/router/scheduler.rs#L64-L67) we are adding the connection_id to `readyqueue` if call returned a `Some(T)`. 

And there can be a case where this `reschedule` is called multiple times in [this loop](https://github.com/bytebeamio/rumqtt/blob/fe8a8e3c12274846965f3bbcf504706cac654d6e/rumqttd/src/router/routing.rs#LL645C1-L645C21)

Reply:
yup